### PR TITLE
feat: Spell 2026-04-09 (DEV-1522)

### PR DESCRIPTION
**Pull request by TheMj0ln1r**
# YYYY-MM-DD Spark Spell

## Forum Post
< Forum post URL >

## Dependencies on Sky Core spell
- List of items (or None)

## Addresses

| Variable name           | Address    | Network   | Source of truth                 |
| ---                     | ---        | ---       | ---                             |
| `PT_SUSDE_MAR26`        | `0x123...` | `mainnet` | specified in forum post         |
| `PT_SUSDE_MAR26_ORACLE` | `0x456...` | `mainnet` | provided in TG chat with Pendle |

## Notes for Reviewers
- Pay extra attention to: ...
- Extra feedback wanted on: ...

## Conditions for Spell Deployment
- [x] All spell-specific tests are passing.
- [x] All E2E tests are passing.
- [x] All newly deployed addresses have been reviewed and added to the spark-address-registry in a PR that has been merged.
- [x] All newly deployed addresses have been added to the Immunefi bug bounty program.
- [x] There is complete test coverage of all changes in the spell, including any relevant edge cases.
- [x] At least three approvals have been given from reviewers.
- [x] CODE_RABBIT.md comment has been posted, AI review performed.
- [x] Octane review performed.

## Conditions for Spell Handoff to Sky
- [x] All spells payloads have been deployed and verified on Etherscan.
- [x] All spell-specific tests are passing (including spell bytecode verification).
- [x] All E2E tests are passing.
- [x] Spell caster is passing in CI.
- [x] At least three approvals have been given from reviewers on deployed spell.
- [x] Comment from at least three reviewers confirming address of mainnet spell payload.

## Conditions for Spell Merge
- [ ] Spell handoff performed to Sky team on Discord.
- [ ] Spell handoff performed to Sky team on Signal.
- [ ] Spell address confirmed to be correct in sky-ecosystem/spells-mainnet repo open PR through comments by at least three reviewers.
- [ ] Sky deployed spell confirmed through a comment in this PR to contain correct Spark payload address by at least three reviewers.

## Summary by CodeRabbit

* **New Features**
  * Added proposal 20260409 implementing rate limit configuration updates across Ethereum, Arbitrum One, and Base networks
  * Increased deposit caps for Spark Savings Vaults (spUSDC, spUSDT, spETH)

* **Tests**
  * Added comprehensive test coverage for proposal execution, rate limit state validation, and vault deposit cap boundary testing across all networks

### CODE REVIEW

File: src/proposals/20260409/Spell_20260409.t.sol [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+
+    function setUp() public override {
+        super.setUp();
+
+        // chainData[ChainIdUtils.Ethereum()].payload    = ;
+        // chainData[ChainIdUtils.ArbitrumOne()].payload = ;
+        // chainData[ChainIdUtils.Base()].payload        = ;
+    }
+
+    function test_officeHours() external onChain(ChainIdUtils.Ethereum()) {
```
Comment (lucas-manuel) [inline src/proposals/20260409/Spell_20260409.t.sol]: Move to SpellTests

File: src/proposals/20260409/SparkEthereum_20260409.sol [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        _configureAaveToken({
+            token        : SparkLend.USDT_SPTOKEN,
+            depositMax   : 500_000_000e6,
+            depositSlope : 2_000_000_000e6 / uint256(1 days)
+        });
+
+        // 7. Increase Rate Limit for Maple syrupUSDT
+        bytes32 SYRUP_USDT_DEPOSIT_KEY  = RateLimitHelpers.makeAddressKey(erc4626DepositKey,  Ethereum.SYRUP_USDT);
+        bytes32 SYRUP_USDT_WITHDRAW_KEY = RateLimitHelpers.makeAddressKey(erc4626WithdrawKey, Ethereum.SYRUP_USDT);
+        bytes32 SYRUP_USDT_REDEEM_KEY   = RateLimitHelpers.makeAddressKey(mapleRedeemKey,     Ethereum.SYRUP_USDT);
```
Comment (lucas-manuel) [inline src/proposals/20260409/SparkEthereum_20260409.sol:109]: Leave spacing between comments like this everywhere in payloads: 

```solidity
        // 6. Increase Rate Limit for SparkLend USDT
        
        _configureAaveToken({
            token        : SparkLend.USDT_SPTOKEN,
            depositMax   : 500_000_000e6,
            depositSlope : 2_000_000_000e6 / uint256(1 days)
        });

        // 7. Increase Rate Limit for Maple syrupUSDT

        bytes32 SYRUP_USDT_DEPOSIT_KEY  = RateLimitHelpers.makeAddressKey(erc4626DepositKey,  Ethereum.SYRUP_USDT);
        bytes32 SYRUP_USDT_WITHDRAW_KEY = RateLimitHelpers.makeAddressKey(erc4626WithdrawKey, Ethereum.SYRUP_USDT);
        bytes32 SYRUP_USDT_REDEEM_KEY   = RateLimitHelpers.makeAddressKey(mapleRedeemKey,     Ethereum.SYRUP_USDT);
```

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: src/proposals/20260409/SparkEthereum_20260409.sol [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        bytes32 erc4626WithdrawKey = almController.LIMIT_4626_WITHDRAW();
+        bytes32 mapleRedeemKey     = almController.LIMIT_MAPLE_REDEEM();
+        bytes32 transferKey        = almController.LIMIT_ASSET_TRANSFER();
+
+        // 1. Deactivate Fluid sUSDS
+
+        bytes32 FLUID_SUSDS_DEPOSIT_KEY  = RateLimitHelpers.makeAddressKey(erc4626DepositKey,  Ethereum.FLUID_SUSDS);
+        bytes32 FLUID_SUSDS_WITHDRAW_KEY = RateLimitHelpers.makeAddressKey(erc4626WithdrawKey, Ethereum.FLUID_SUSDS);
+
+        IRateLimits(rateLimits).setRateLimitData(FLUID_SUSDS_DEPOSIT_KEY,  0, 0);
```
Comment (lucas-manuel) [inline src/proposals/20260409/SparkEthereum_20260409.sol]: Don't cast anywhere here

File: src/proposals/20260409/Spell_20260409.t.sol [hunk 4]
```diff
@@ -0,0 +1,457 @@
...
+    function test_ETHEREUM_sll_curveWeethWethng() external onChain(ChainIdUtils.Ethereum()) {
+        SparkLiquidityLayerContext memory ctx = _getSparkLiquidityLayerContext();
+        MainnetController controller = MainnetController(ctx.controller);
+
+        bytes32 swapKey = RateLimitHelpers.makeAddressKey(
+            controller.LIMIT_CURVE_SWAP(),
+            Ethereum.CURVE_WEETHWETHNG
+        );
+
+        _assertRateLimit(swapKey, 100e18, 1_000e18 / uint256(1 days));
```
Comment (lucas-manuel) [inline src/proposals/20260409/Spell_20260409.t.sol]: Need to assert rate limits of deposit and withdraw to ensure that they did not change

File: src/proposals/20260409/Spell_20260409.t.sol [hunk 5]
```diff
@@ -0,0 +1,457 @@
...
+    function test_ETHEREUM_sll_curveSusdsUsdt() external onChain(ChainIdUtils.Ethereum()) {
+        SparkLiquidityLayerContext memory ctx = _getSparkLiquidityLayerContext();
+        MainnetController controller = MainnetController(ctx.controller);
+
+        bytes32 swapKey = RateLimitHelpers.makeAddressKey(
+            controller.LIMIT_CURVE_SWAP(),
+            Ethereum.CURVE_SUSDSUSDT
+        );
+
+        _assertRateLimit(swapKey, 5_000_000e18, 100_000_000e18 / uint256(1 days));
```
Comment (lucas-manuel) [inline src/proposals/20260409/Spell_20260409.t.sol]: Need to assert rate limits of deposit and withdraw to ensure that they did not change

File: src/proposals/20260409/SparkArbitrumOne_20260409.sol [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        ForeignController almController = ForeignController(Arbitrum.ALM_CONTROLLER);
+        IRateLimits       rateLimits    = IRateLimits(Arbitrum.ALM_RATE_LIMITS);
+
+        // 1. Deactivate Aave USDC
+
+        bytes32 ATOKEN_USDC_DEPOSIT_KEY  = RateLimitHelpers.makeAddressKey(almController.LIMIT_AAVE_DEPOSIT(),  Arbitrum.ATOKEN_USDC);
+        bytes32 ATOKEN_USDC_WITHDRAW_KEY = RateLimitHelpers.makeAddressKey(almController.LIMIT_AAVE_WITHDRAW(), Arbitrum.ATOKEN_USDC);
+
+        IRateLimits(rateLimits).setRateLimitData(ATOKEN_USDC_DEPOSIT_KEY,  0, 0);
+        IRateLimits(rateLimits).setRateLimitData(ATOKEN_USDC_WITHDRAW_KEY, 0, 0);
```
Comment (lucas-manuel) [inline src/proposals/20260409/SparkArbitrumOne_20260409.sol]: Rm casting

File: src/proposals/20260409/SparkBase_20260409.sol [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        ForeignController almController = ForeignController(Base.ALM_CONTROLLER);
+        IRateLimits       rateLimits    = IRateLimits(Base.ALM_RATE_LIMITS);
+
+        // 1. Deactivate Aave USDC
+
+        bytes32 ATOKEN_USDC_DEPOSIT_KEY  = RateLimitHelpers.makeAddressKey(almController.LIMIT_AAVE_DEPOSIT(),  Base.ATOKEN_USDC);
+        bytes32 ATOKEN_USDC_WITHDRAW_KEY = RateLimitHelpers.makeAddressKey(almController.LIMIT_AAVE_WITHDRAW(), Base.ATOKEN_USDC);
+
+        IRateLimits(rateLimits).setRateLimitData(ATOKEN_USDC_DEPOSIT_KEY,  0, 0);
+        IRateLimits(rateLimits).setRateLimitData(ATOKEN_USDC_WITHDRAW_KEY, 0, 0);
```
Comment (lucas-manuel) [inline src/proposals/20260409/SparkBase_20260409.sol]: Rm casting

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: src/proposals/20260409/Spell_20260409.t.sol
Comment (lucas-manuel) [inline src/proposals/20260409/Spell_20260409.t.sol]: Need to do same deposit/withdraw rate limit checks here

File: src/proposals/20260409/SparkEthereum_20260409.sol [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+ * Vote:
+ *
+ */
+contract SparkEthereum_20260409 is SparkPayloadEthereum {
+
+    address internal constant ANCHORAGE_USAT_USDT = 0x49506C3Aa028693458d6eE816b2EC28522946872;
+
+    ISparkVaultV2Like internal constant spEth  = ISparkVaultV2Like(Ethereum.SPARK_VAULT_V2_SPETH);
+    ISparkVaultV2Like internal constant spUsdc = ISparkVaultV2Like(Ethereum.SPARK_VAULT_V2_SPUSDC);
+    ISparkVaultV2Like internal constant spUsdt = ISparkVaultV2Like(Ethereum.SPARK_VAULT_V2_SPUSDT);
```
Comment (supercontracts) [inline src/proposals/20260409/SparkEthereum_20260409.sol:37]: These can be added in the _postExecute function directly instead of declaring outside

File: src/proposals/20260409/Spell_20260409.t.sol [hunk 9]
```diff
@@ -0,0 +1,471 @@
...
+        _testSparkVaultDepositCapBoundary({
+            vault:              usdtVault,
+            depositCap:         10_000_000_000e6,
+            expectedMaxDeposit: 9_189_811_901.398293e6
+        });
+        _testSparkVaultDepositCapBoundary({
+            vault:              ethVault,
+            depositCap:         1_000_000e18,
+            expectedMaxDeposit: 983_634.985416955036945334e18
+        });
```
Comment (supercontracts) [inline src/proposals/20260409/Spell_20260409.t.sol:411]: Align all like this 
```
_testSparkVaultDepositCapBoundary({
    vault              : usdcVault,
    depositCap         : 10_000_000_000e6,
    expectedMaxDeposit : 9_588_692_932.754565e6
});
```

File: src/proposals/20260409/SparkBase_20260409.sol [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+import { SparkPayloadBase } from "src/SparkPayloadBase.sol";
+
+/**
+ * @title  April 09, 2026 Spark Base Proposal
+ * @author Phoenix Labs
+ * @notice Spark Liquidity Layer:
+ *         - Deactivate Aave USDC and Fluid sUSDS
+ * Forum:  https://forum.skyeco.com/t/april-9-2026-proposed-changes-to-spark-for-upcoming-spell/27804
+ * Vote:
```
Comment (supercontracts) [inline src/proposals/20260409/SparkBase_20260409.sol]: Add vote links - https://snapshot.box/#/s:sparkfi.eth

File: src/proposals/20260409/SparkArbitrumOne_20260409.sol [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+import { SparkPayloadArbitrumOne } from "src/SparkPayloadArbitrumOne.sol";
+
+/**
+ * @title  April 09, 2026 Spark Arbitrum Proposal
+ * @author Phoenix Labs
+ * @notice Spark Liquidity Layer:
+ *         - Deactivate Aave USDC and Fluid sUSDS
+ * Forum:  https://forum.skyeco.com/t/april-9-2026-proposed-changes-to-spark-for-upcoming-spell/27804
+ * Vote:
```
Comment (supercontracts) [inline src/proposals/20260409/SparkArbitrumOne_20260409.sol]: Add vote links - https://snapshot.box/#/s:sparkfi.eth

File: src/proposals/20260409/SparkEthereum_20260409.sol [hunk 12]
```diff
@@ -0,0 +1,220 @@
...
+ *         - Deactivate Fluid sUSDS, Aave Prime USDS, Aave Core USDS, Aave Core USDC
+ *         - Configure Aave Core USDT, SparkLend USDT, SparkLend ETH
+ *         - Configure Maple syrupUSDT
+ *         - Configure Morpho Blue Chip USDT Vault
+ *         - Configure Curve weETH/WETH-ng and sUSDS/USDT
+ *         - Update Anchorage USDT and USAT transfer asset limits
+ *         Spark Savings:
+ *         - Increase spUSDT, spUSDC, spETH deposit caps
+ * Forum:  https://forum.skyeco.com/t/april-9-2026-proposed-changes-to-spark-for-upcoming-spell/27804
+ * Vote:
```
Comment (supercontracts) [inline src/proposals/20260409/SparkEthereum_20260409.sol]: Add vote links - https://snapshot.box/#/s:sparkfi.eth

File: src/proposals/20260409/Spell_20260409.t.sol [hunk 13]
```diff
@@ -0,0 +1,471 @@
...
+
+        _assertRateLimit(depositKey,  10_000_000e6,      1_000_000_000e6 / uint256(1 days));
+        _assertRateLimit(withdrawKey, type(uint256).max, 0);
+
+        _executeAllPayloadsAndBridges();
+
+        _assertRateLimit(depositKey,  100_000_000e6,     1_000_000_000e6 / uint256(1 days));
+        _assertRateLimit(withdrawKey, type(uint256).max, 0);
+
+        _testAaveIntegration(E2ETestParams(ctx, Ethereum.ATOKEN_CORE_USDT, 1_000_000e6, depositKey, withdrawKey, 10));
```
Comment (supercontracts) [inline src/proposals/20260409/Spell_20260409.t.sol]: put depositAmount here as 100_000_000e6

File: src/proposals/20260409/Spell_20260409.t.sol [hunk 14]
```diff
@@ -0,0 +1,471 @@
...
+
+        _assertRateLimit(depositKey,  250_000_000e6,     2_000_000_000e6 / uint256(1 days));
+        _assertRateLimit(withdrawKey, type(uint256).max, 0);
+
+        _executeAllPayloadsAndBridges();
+
+        _assertRateLimit(depositKey,  500_000_000e6,     2_000_000_000e6 / uint256(1 days));
+        _assertRateLimit(withdrawKey, type(uint256).max, 0);
+
+        _testAaveIntegration(E2ETestParams(ctx, SparkLend.USDT_SPTOKEN, 1_000_000e6, depositKey, withdrawKey, 10));
```
Comment (supercontracts) [inline src/proposals/20260409/Spell_20260409.t.sol]: Put depositAmount as 500_000_000e6 instead of 1_000_000e6

File: src/proposals/20260409/Spell_20260409.t.sol [hunk 15]
```diff
@@ -0,0 +1,471 @@
...
+
+        _assertRateLimit(depositKey,  50_000e18,         10_000e18 / uint256(1 days));
+        _assertRateLimit(withdrawKey, type(uint256).max, 0);
+
+        _executeAllPayloadsAndBridges();
+
+        _assertRateLimit(depositKey,  50_000e18,         250_000e18 / uint256(1 days));
+        _assertRateLimit(withdrawKey, type(uint256).max, 0);
+
+        _testAaveIntegration(E2ETestParams(ctx, SparkLend.WETH_SPTOKEN, 100e18, depositKey, withdrawKey, 10));
```
Comment (supercontracts) [inline src/proposals/20260409/Spell_20260409.t.sol]: Put deposit amount as 50_000e18

File: src/proposals/20260409/Spell_20260409.t.sol [hunk 16]
```diff
@@ -0,0 +1,471 @@
...
+        _executeAllPayloadsAndBridges();
+
+        _assertRateLimit(depositKey, 50_000_000e6, 100_000_000e6 / uint256(1 days));
+        _assertUnlimitedRateLimit(redeemKey);
+        _assertUnlimitedRateLimit(withdrawKey);
+
+        _testMapleIntegration(MapleE2ETestParams({
+            ctx           : ctx,
+            vault         : Ethereum.SYRUP_USDT,
+            depositAmount : 1_000_000e6,
```
Comment (supercontracts) [inline src/proposals/20260409/Spell_20260409.t.sol]: depositAmount: 50_000_000e6

File: src/proposals/20260409/Spell_20260409.t.sol [hunk 17]
```diff
@@ -0,0 +1,471 @@
...
+        _assertRateLimit(swapKey, 1_000e18, 50_000e18 / uint256(1 days));
+
+        assertEq(controller.maxSlippages(Ethereum.CURVE_WEETHWETHNG), 0.9975e18);
+
+        _testCurveSwapIntegration(CurveSwapE2ETestParams({
+            ctx        : ctx,
+            pool       : Ethereum.CURVE_WEETHWETHNG,
+            asset0     : Ethereum.WETH,
+            asset1     : Ethereum.WEETH,
+            swapAmount : 10e18,
```
Comment (supercontracts) [inline src/proposals/20260409/Spell_20260409.t.sol]: use 1_000e18 as swapAmount

File: src/proposals/20260409/Spell_20260409.t.sol [hunk 18]
```diff
@@ -0,0 +1,471 @@
...
+    function test_ETHEREUM_sll_curveWeethWethng() external onChain(ChainIdUtils.Ethereum()) {
+        SparkLiquidityLayerContext memory ctx = _getSparkLiquidityLayerContext();
+        MainnetController controller = MainnetController(ctx.controller);
+
+        bytes32 swapKey = RateLimitHelpers.makeAddressKey(
+            controller.LIMIT_CURVE_SWAP(),
+            Ethereum.CURVE_WEETHWETHNG
+        );
+
+        _assertRateLimit(swapKey, 100e18, 1_000e18 / uint256(1 days));
```
Comment (supercontracts) [inline src/proposals/20260409/Spell_20260409.t.sol]: check deposit and withdraw keys remain unchanged

File: src/proposals/20260409/Spell_20260409.t.sol [hunk 19]
```diff
@@ -0,0 +1,471 @@
...
+        _executeAllPayloadsAndBridges();
+
+        _assertRateLimit(transferKey, 50_000_000e6, 250_000_000e6 / uint256(1 days));
+
+        _testTransferAssetIntegration(TransferAssetE2ETestParams({
+            ctx            : _getSparkLiquidityLayerContext(),
+            asset          : Ethereum.USDT,
+            destination    : ANCHORAGE_USAT_USDT,
+            transferKey    : transferKey,
+            transferAmount : 5_000_000e6
```
Comment (supercontracts) [inline src/proposals/20260409/Spell_20260409.t.sol]: use 50_000_000e6 as transferAmount

File: src/proposals/20260409/Spell_20260409.t.sol [hunk 20]
```diff
@@ -0,0 +1,471 @@
...
+        _executeAllPayloadsAndBridges();
+
+        _assertRateLimit(transferKey, 50_000_000e6, 250_000_000e6 / uint256(1 days));
+
+        _testTransferAssetIntegration(TransferAssetE2ETestParams({
+            ctx            : _getSparkLiquidityLayerContext(),
+            asset          : Ethereum.USAT,
+            destination    : ANCHORAGE_USAT_USDT,
+            transferKey    : transferKey,
+            transferAmount : 5_000_000e6
```
Comment (supercontracts) [inline src/proposals/20260409/Spell_20260409.t.sol]: use 50_000_000e6 as transferAmount

**Review by supercontracts — COMMENTED** [review]
changes

**Review by supercontracts — CHANGES_REQUESTED** [review]

File: src/proposals/20260409/SparkEthereum_20260409.sol (hunk 8 above)
Comment (TheMj0ln1r) [reply src/proposals/20260409/SparkEthereum_20260409.sol:37]: It was stack-to-deep, choosing constants over additional helper functions.

File: src/test-harness/SparkLiquidityLayerTests.sol [hunk 21]
```diff
@@ -575,16 +576,28 @@ abstract contract SparkLiquidityLayerTests is SpellRunner {
...
     function _handleMorphoFees(E2ETestParams memory p) internal {
-        // If the feeRecipient is set, the vault will accrue fees into the ALMProxy during e2e test
+        // If the vault accrues performance/management fees into the ALMProxy during e2e test
         // deposit, causing unexpected behavior. This is a workaround to avoid this.
         // Using 100 instead of 1 to avoid share validation issue.
+        // Checks both feeRecipient() (MetaMorpho V1) and performanceFee() (V2) interfaces.
+
+        bool hasFees;
+
         try IMetaMorpho(p.vault).feeRecipient() {
```
Comment (certora-spark-review) [inline src/test-harness/SparkLiquidityLayerTests.sol]: `feeRecipient` could be zero as well if not set. For consistency, consider adding below validation
```solidity
try IMetaMorpho(p.vault).feeRecipient() returns (address recipient) {
    hasFees = recipient != address(0);
}
```

File: src/test-harness/SparkLiquidityLayerTests.sol [hunk 22]
```diff
@@ -575,16 +576,28 @@ abstract contract SparkLiquidityLayerTests is SpellRunner {
...
+            hasFees = true;
+        } catch {
+            try IMorphoVaultV2Like(p.vault).performanceFee() returns (uint96 fee) {
```
Comment (certora-spark-review) [inline src/test-harness/SparkLiquidityLayerTests.sol]: Consider checking `managementFee` as well.

**Review by supercontracts — DISMISSED** [review]

File: src/proposals/20260409/Spell_20260409.t.sol [hunk 23]
```diff
@@ -0,0 +1,489 @@
...
+
+        _assertRateLimit(depositKey,  50_000_000e6,      1_000_000_000e6 / uint256(1 days));
+        _assertRateLimit(withdrawKey, type(uint256).max, 0);
+
+        _executeAllPayloadsAndBridges();
+
+        _assertRateLimit(depositKey,  100_000_000e6,     1_000_000_000e6 / uint256(1 days));
+        _assertRateLimit(withdrawKey, type(uint256).max, 0);
+
+        _testERC4626Integration(E2ETestParams(ctx, Ethereum.MORPHO_VAULT_V2_USDT, 1_000_000e6, depositKey, withdrawKey, 10));
```
Comment (lucas-manuel) [inline src/proposals/20260409/Spell_20260409.t.sol]: 100_000_000e6

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
Small changes, still need to wait for deposit cap updates

- [x]  Check forum post link
- [x]  Check description for all payloads
- [x]  Check poll links
- [x]  Check spark-address-registry commit matches current HEAD
- [x]  SLL Ethereum Rate Limits
    - SLL rate limit set to zero (Fluid sUSDS) (DONE)
        - [x]  Check integration address against forum post
        - [x]  Check maxAmount and slope set to zero for deposit
        - [x]  Check maxAmount and slope set to zero for withdraw
        - [x]  Check rate limit before and after state in a test
    - SLL rate limit set to zero (Aave Prime USDS) (DONE)
        - [x]  Check integration address against forum post
        - [x]  Check maxAmount and slope set to zero for deposit
        - [x]  Check maxAmount and slope set to zero for withdraw
        - [x]  Check rate limit before and after state in a test
    - SLL rate limit set to zero (Aave Core USDS) (DONE)
        - [x]  Check integration address against forum post
        - [x]  Check maxAmount and slope set to zero for deposit
        - [x]  Check maxAmount and slope set to zero for withdraw
        - [x]  Check rate limit before and after state in a test
    - SLL rate limit set to zero (Aave Core USDC) (DONE)
        - [x]  Check integration address against forum post
        - [x]  Check maxAmount and slope set to zero for deposit
        - [x]  Check maxAmount and slope set to zero for withdraw
        - [x]  Check rate limit before and after state in a test
    - AAVE-style SLL rate limit adjustment (Aave Core USDT) (DONE)
        - [x]  Check aToken address against forum post
        - [x]  Check maxAmounts against forum post
        - [x]  Check slopes against forum post
        - [x]  Verify the precision of the rate limit against the underlying asset
        - [x]  Check rate limit before and after state in a test
        - [x]  Ensure that helper function is used in spell
        - [x]  Ensure that `_testAaveIntegration` is run within test
    - AAVE-style SLL rate limit adjustment (SparkLend USDT) (DONE)
        - [x]  Check aToken address against forum post
        - [x]  Check maxAmounts against forum post
        - [x]  Check slopes against forum post
        - [x]  Verify the precision of the rate limit against the underlying asset
        - [x]  Check rate limit before and after state in a test
        - [x]  Ensure that helper function is used in spell
        - [x]  Ensure that `_testAaveIntegration` is run within test
    - Maple-style SLL rate limit adjustment (Maple syrupUSDT) (DONE)
        - [x]  Check pool address against forum post
        - [x]  Check maxAmounts against forum post
        - [x]  Check slopes against forum post
        - [x]  Check that its the requestRedeem rate limit not the withdraw rate limit thats set
        - [x]  Verify the precision of the rate limit against the underlying asset
        - [x]  Check rate limit before and after state in a test
        - [x]  Ensure that helper function is used in spell
        - [x]  Ensure that `_testMapleIntegration` is run within test
    - Morpho-style SLL rate limit adjustment (Morpho Blue Chip USDT) (DONE)
        - [x]  Check vault address against forum post
        - [x]  Check maxAmounts against forum post
        - [x]  Check slopes against forum post
        - [x]  Verify the precision of the rate limit against the underlying asset
        - [x]  Check rate limit before and after state in a test
        - [x]  Ensure that helper function is used in spell
        - [x]  Ensure that `_testERC4626Integration` is run within test
    - AAVE-style SLL rate limit adjustment (SparkLend ETH) (DONE)
        - [x]  Check aToken address against forum post
        - [x]  Check maxAmounts against forum post
        - [x]  Check slopes against forum post
        - [x]  Verify the precision of the rate limit against the underlying asset
        - [x]  Check rate limit before and after state in a test
        - [x]  Ensure that helper function is used in spell
        - [x]  Ensure that `_testAaveIntegration` is run within test
    - Curve-style SLL rate limit adjustment (Curve weETH-ng) (DONE)
        - [x]  Check pool address against forum post
        - [x]  Check maxAmounts against forum post
        - [x]  Check slopes against forum post
        - [x]  Verify the precision of the rate limit against the underlying asset
        - [x]  Check rate limit before and after state in a test
        - [x]  Ensure that helper function is used in spell
        - [x]  Ensure that `_testCurveIntegration` is run within test
    - Curve-style SLL rate limit adjustment (Curve sUSDS/USDT) (DONE)
        - [x]  Check pool address against forum post
        - [x]  Check maxAmounts against forum post
        - [x]  Check slopes against forum post
        - [x]  Verify the precision of the rate limit against the underlying asset
        - [x]  Check rate limit before and after state in a test
        - [x]  Ensure that helper function is used in spell
        - [x]  Ensure that `_testCurveIntegration` is run within test
    - TransferAsset SLL rate limit setting (Anchorage USDT) (DONE)
        - [x]  Check asset address against forum post
        - [x]  Check recipient address against forum post
        - [x]  Check maxAmount against forum post
        - [x]  Check slope against forum post
        - [x]  Verify the precision of the rate limit against the underlying asset
        - [x]  Check rate limit before and after state in a test
        - [x]  Ensure that `_testTransferAssetIntegration` is run with new integration in spell tests
    - TransferAsset SLL rate limit setting (Anchorage USAT) (DONE)
        - [x]  Check asset address against forum post
        - [x]  Check recipient address against forum post
        - [x]  Check maxAmount against forum post
        - [x]  Check slope against forum post
        - [x]  Verify the precision of the rate limit against the underlying asset
        - [x]  Check rate limit before and after state in a test
        - [x]  Ensure that `_testTransferAssetIntegration` is run with new integration in spell tests
- [x]  SLL Arbitrum Rate Limits
    - SLL rate limit set to zero (Fluid Arbitrum sUSDS)
        - [x]  Check integration address against forum post
        - [x]  Check maxAmount and slope set to zero for deposit
        - [x]  Check maxAmount and slope set to zero for withdraw
        - [x]  Check rate limit before and after state in a test
    - SLL rate limit set to zero (Aave Arbitrum USDC)
        - [x]  Check integration address against forum post
        - [x]  Check maxAmount and slope set to zero for deposit
        - [x]  Check maxAmount and slope set to zero for withdraw
        - [x]  Check rate limit before and after state in a test
- [x]  SLL Base Rate Limits
    - SLL rate limit set to zero (Aave Base USDC)
        - [x]  Check integration address against forum post
        - [x]  Check maxAmount and slope set to zero for deposit
        - [x]  Check maxAmount and slope set to zero for withdraw
        - [x]  Check rate limit before and after state in a test
    - SLL rate limit set to zero (Fluid Base sUSDS)
        - [x]  Check integration address against forum post
        - [x]  Check maxAmount and slope set to zero for deposit
        - [x]  Check maxAmount and slope set to zero for withdraw
        - [x]  Check rate limit before and after state in a test
- [ ]  Spark Savings - Raise Deposit Caps for spUSDC, spUSDT, and spETH
    - SparkVaultV2 deposit cap increase (spUSDC)
        - [x]  Check the address of the vault
        - [x]  Check the value of the new deposit cap
        - [x]  Check the decimals of the asset, ensuring correct precision
        - [x]  Review the test
            - [x]  Deposit cap before spell
            - [x]  Deposit cap after spell
            - [x]  Use `_testSparkVaultDepositCapBoundary` to check deposit cap is in effect after spell
    - SparkVaultV2 deposit cap increase (spUSDT)
        - [x]  Check the address of the vault
        - [x]  Check the value of the new deposit cap
        - [x]  Check the decimals of the asset, ensuring correct precision
        - [x]  Review the test
            - [x]  Deposit cap before spell
            - [x]  Deposit cap after spell
            - [x]  Use `_testSparkVaultDepositCapBoundary` to check deposit cap is in effect after spell
    - SparkVaultV2 deposit cap increase (spETH)
        - [x]  Check the address of the vault
        - [x]  Check the value of the new deposit cap
        - [x]  Check the decimals of the asset, ensuring correct precision
        - [x]  Review the test
            - [x]  Deposit cap before spell
            - [x]  Deposit cap after spell
            - [x]  Use `_testSparkVaultDepositCapBoundary` to check deposit cap is in effect after spell
- [x]  Withdraw reserves
    - [x]  Review SparkPayloadEthereum code for claiming all reserves
    - [x]  Ensure stablecoins go to ALMProxy and everything else goes to Liquidation multisig
    - [x]  Review that test decrements and increments funds from correct addresses for all reserves

**Review by lucas-manuel — APPROVED** [review]

**Review by lucas-manuel — DISMISSED** [review]
Approving spell for deployment.

**Review by certora-spark-review — DISMISSED** [review]
Approving the spell for deployment.

**Review by certora-spark-review — APPROVED** [review]
Confirming and approving the spell deployment. Ethereum payload is deployed at `0xFa5fc020311fCC1A467FEC5886640c7dD746deAa`

**Review by supercontracts — APPROVED** [review]
Approving the payload at `0xFa5fc020311fCC1A467FEC5886640c7dD746deAa`

**Review by lucas-manuel — APPROVED** [review]
Approving spell deployment at `0xFa5fc020311fCC1A467FEC5886640c7dD746deAa`

**Review by lucas-manuel — APPROVED** [review]
